packages/codegen-ui-react/lib/forms/react-form-renderer.ts

```diff
@@ -447,6 +448,17 @@ export abstract class ReactFormTemplateRenderer extends StudioTemplateRenderer<
           ['id', lowerCaseDataTypeName],
         ),
       );
+
+      // Build effects to grab nested models off target record for relationships
+      Object.entries(formMetadata.fieldConfigs).forEach(([key, value]) => {
+        if (value.relationship) {
```

Comment:
There's a place in this file that already iterates through fieldConfigs if you want to collapse this down there

Reply:
I'll move this into the other effect as suggested by @zchenwei , so ill have iterate and pass the field configs into the `buildUpdateDatastoreQuery` function

Reply:
Oh yeah, I think a conditional is required to check the the relationship type. For `hasMany` and `manyToMany`, the data lazy loading part is a bit different, it needs to call `toArray()`.

Reply:
Should I check for `belongsTo` and `hasOne` in that check then?